### ISSUE TITLE
Code to replicate figure 1 (Accuracy score progression of OLMo-7B) from Olmo paper

### ISSUE THREAD

**Issue by luffycodes** [allenai/OLMo#825]
### ❓ The question

hi i was wondering if there is a code base to replicate figure 1 (caption: Accuracy score progression of OLMo-7B on 8 core end-tasks score from Catwalk evaluation suite) from [omlo paper](https://aclanthology.org/2024.acl-long.841.pdf)? i wanted to perform similar analysis for larger olmo models. thank you in advance!

**Comment by aman-17** [comment]
Hey @luffycodes, not sure if we have this code, need to go lot of commits back and check it. In the mean while, I can provide you with training logs.
[OLMo 1B](https://wandb.ai/ai2-llm/OLMo-1B/reports/OLMo-1B--Vmlldzo2NzY1Njk1) report. 
[OLMo 7B](https://wandb.ai/ai2-llm/OLMo-7B/reports/OLMo-7B--Vmlldzo2NzQyMzk5) report.

**Comment by luffycodes** [comment]
thanks ! wanted to run the analysis for bigger models actually. any tips
how to code it up? few pointers would be great - just which library to use
for eval will do the trick?

also does it make sense to fine-tune the model checkpoints first on a small
instruction tune dataset before analyzing their performance? in the paper
after fine-tuning the final model, its performance on mmlu takes a steep
boost. fine-tuning might take a bit of time but i was wondering can be a
necessary step before gauging the intermittent checkpoint's performance. if
so can you also recommend the 1) library for fine-tuning, 2) any
recommended instruction tuning dataset and 3) any script which lists the
hyper parameters. thank you so much for your help!

**Comment by dirkgr** [comment]
Those graphs from the paper were made with [Catwalk](https://github.com/allenai/catwalk). You can run that yourself, on any huggingface-compatible model (@aman-17 pointed to a script above, but you can also just run Catwalk by itself).

The boost at the end doesn't come from fine-tuning, but it comes from annealing. The difference is a little subtle. For fine-tuning, we would train on the training sets for each of the tasks and then evaluate, but we did not do that. Instead, at the end of training we switch the learning rate schedule to quickly go to zero over 1000 more steps. This is what boosts the performance so much at the end, and it is not very expensive.

When we made those graphs, we did not anneal each of the checkpoints before evaluating them, but you certainly could do that. It would give you a qualitatively different result, maybe a "truer" representation of the quality of the model at that point.

**Comment by luffycodes** [comment]
Thank you for the response @dirkgr. Is there anything special of the training dataset used in the last 1000 steps if I were to replicate the process for the checkpoints. Thank you in advance !
